kernel/dw1000.c

```diff
@@ -1412,29 +1410,23 @@ static void dw1000_timestamp(struct dw1000 *dw,
 			     const union dw1000_timestamp *time,
 			     struct skb_shared_hwtstamps *hwtstamps)
 {
-	uint64_t cc;
-	uint32_t cc_frac;
-	uint64_t ns;
-	uint64_t ns_frac;
+	struct hires_counter *tc = &dw->ptp.tc;
+	struct timehires stamp;
+	cycle_t cc;
+
+	/* Get 5-byte timestamp */
+	cc = le64_to_cpu(time->cc) & 0xFFFFFFFFFFULL;
```

Comment:
The mask is unnecessary; the most significant bytes are memset() to zero and never subsequently overwritten